k8_cortx_cloud/cortx-cloud-3rd-party-pkg/templates/job-symlink-block-devices.yaml.template

```
@@ -0,0 +1,99 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: symlink-block-devices-${NODE_SHORT_NAME}
+  labels:
+    cortx.io/task: symlink-block-devices
+    kubernetes.io/hostname: ${NODE_NAME}
+spec:
+  backoffLimit: 1
+  template:
+    metadata:
+      labels:
+        cortx.io/task: symlink-block-devices
+    spec:
+      nodeSelector:
+        kubernetes.io/hostname: ${NODE_NAME}
+      restartPolicy: Never
+      volumes:
+      - name: node-devices
+        hostPath:
+          path: /dev
+          type: Directory
+      containers:
+      - name: symlink-block-devices-task
+        image: ${CORTX_IMAGE}
+        command: 
+        - "bash"
+        - "-c"
+        - |
+          #!/bin/bash
+          printf "#############################################################################\n"
+          printf "# Creating block device symlinks for %s\n" ${JOB_NODE_NAME}
+          printf "#############################################################################\n"
+          printf "Symlink path separator:\t ${JOB_SYMLINK_PATH_SEPARATOR}\n"
+          printf "Device paths:\t\t${JOB_DEVICE_PATHS}\n"
+          printf "Creating symlinks...\n"
+
+          # If symlink sub-directory does not exist, create it
+          if [[ ! -d /job/dev/${JOB_SYMLINK_PATH_SEPARATOR} ]]; then
+            mkdir -p /job/dev/${JOB_SYMLINK_PATH_SEPARATOR}
+          fi
+
+          ERROR_COUNT=0
+
+          IFS=',' read -r -a DEVICE_PATH_ARRAY <<< "${JOB_DEVICE_PATHS}"
+          for INDEX in "${!DEVICE_PATH_ARRAY[@]}"
+          do
+            ITER_DEVICE_PATH=${DEVICE_PATH_ARRAY[${INDEX}]}    
+            printf "\tDevice Path:\t\t\t%s\n" ${ITER_DEVICE_PATH}
+
+            # We use basename here since the `ls -l /dev/disk/by-id` command 
+            # only shows the actual device name and not path elements in its symlink
+            ITER_DEVICE_NAME=$(basename ${ITER_DEVICE_PATH})
+            printf "\tDevice Name:\t\t\t%s\n" ${ITER_DEVICE_NAME}
+            
+            # Look up the block device id based upon host system `/dev/disk/by-id` mappings
+            ITER_BLK_ID=$(ls -l /job/dev/disk/by-id | grep ${ITER_DEVICE_NAME} | awk '{print $9}')
```

Won't this match multiple device paths? In a system with lots of disks you will see repeating elements with matching substrings. E.g. if `/dev/sda` was specified, there might be `/dev/sdaa` and `/dev/sdab`, etc. Or, if there are devices for partitions, `sda1` could match.

Or, I've misunderstood how this is working.

Assuming the symlinks are always relative to `/dev`, maybe match end of the path?
```suggestion
            ITER_BLK_ID=$(ls -l /job/dev/disk/by-id | grep "/${ITER_DEVICE_NAME}$" | awk '{print $9}')
```

---

Yep, good catch!